### PR TITLE
Implement for App Alert

### PR DESCRIPTION
**Pull request by AllenFang**
Commit some useful and important part for reviewing.

**Target**: implement the alert area on `component/app/app.js`.

**Todo**:
- [x] components
- [ ] Unit test

### CODE REVIEW

**Comment by AllenFang** [comment]
Currently, we can get some basic data via this PR

**Comment by AllenFang** [comment]
@CYBAI please have a look on it, this PR have following tasks done, and you can merge it due to it will be easy and helpful for the development of your part.

* `redux-thunk`
* A basic state tree(check `src-react/store.js`) and reducers implementation(check `src-react/reducers`)
* `path` container implementation, now we have an essential entry point for path page

I'm going to open another two PR 

One is for implement the components(`AlertArea`) event, because this PR only have the view(presentation) but I doesn't implement the events(click etc.) yet.

The other is for the testing.

File: src-react/actions/ungit-config.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        }
+        dispatch(receiveUgitConfig(json));
+      })
+      .catch(e => {
+        dispatch(apiError(e.message));
+      });
+  };
+};
+
+function receiveUgitConfig(ungitConfig) {
```
Comment (CYBAI) [inline src-react/actions/ungit-config.js]: I think it's typo in the function name?

`receiveUgitConfig` => receiveU`n`gitConfig

File: src-react/containers/path.js [hunk 2]
```diff
@@ -1,19 +1,41 @@
 import React, { Component } from 'react';
+import { bindActionCreators } from 'redux';
 import { connect } from 'react-redux'
 
+import AlerArea from 'components/alert-area';
```
Comment (CYBAI) [inline src-react/containers/path.js]: typo here for Aler`t`Area

File: src-react/actions/bootstrap.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { fetchUngitConfig } from './ungit-config';
+import { fetchLatestVersion, fetchGitVersion } from './version';
+import { pending } from './common';
+
+export function bootstrap() {
+  return dispatch => {
+    dispatch(pending(4));
```
Comment (CYBAI) [inline src-react/actions/bootstrap.js]: Just want to make my understanding much clearer.
I think the usage of `pending` is to know how many APIs don't finish yet.

If my understanding is correct, according to this `bootstrap` action, there're three `fetch` action, why do you use `4` here?

**Review by CYBAI — CHANGES_REQUESTED** [review]

File: src-react/actions/bootstrap.js (hunk 3 above)
Comment (AllenFang) [reply src-react/actions/bootstrap.js]: Due to ungitconfig action will also fetch userconfig, but after I check the code, there's a condition  to decide if fetching userconfig or not, so I think I need to move a pending call into ungitconfig

It's fixed

File: src-react/containers/path.js (hunk 2 above)
Comment (AllenFang) [reply src-react/containers/path.js]: fixed

File: src-react/actions/ungit-config.js (hunk 1 above)
Comment (AllenFang) [reply src-react/actions/ungit-config.js]: fixed

**Review by CYBAI — APPROVED** [review]

**Pull request merged**